### PR TITLE
Add function to parse previous text log file

### LINKED ISSUE #162: Feature Request:  Add ability to easily subset/filter log files
### Feature Idea

The text log file seems to be formatted in a pretty standard way. With some enhancements, maybe users could parse this text file and grab the info they need (list of package versions for example). Or there could be a feature added to output it into an object that you can more easily subset/filter (like json or rds with a nested list). 

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by parmsam-pfizer**
### Thank you for your Pull Request! 

We have developed a Pull Request template to aid you and our reviewers.  Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the logrx codebase remains robust and consistent.  

### The spirit of logrx

While many packages to facilitate the logging of code already exist in the R ecosystem, it is hard to find a solution that works well for clinical programming applications. Many logging implementations are more implicit and rely on user input to create the log for the execution of a script. While this is useful for logging specific events of an application, in clinical programming a log has a set purpose.

logrx is built around the concept of creating a log for the execution of an R script that provides an overview of what happened as well as the environment that it happened in. We set out to create a flexible logging utility that could provide the necessary information to anyone reviewing the code execution so they can recreate the execution environment and run the code for themselves. Please make sure your Pull Request meets this **spirit of logrx**.

Please check off each taskbox as an acknowledgment that you completed the task. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `dev` branch until you have checked off each task.

- [x] The spirit of logrx is met in your Pull Request
- [x] Check that your Pull Request is targeting the dev branch, Pull Requests to master should use the [Release Pull Request Template](https://github.com/pharmaverse/logrx/blob/main/.github/PULL_REQUEST_TEMPLATE/release.md)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests.  Remember to remove any configured log objects at the end of every test using `log_remove()`.
- [x] Creation/updates to relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Address any updates needed for vignettes and/or templates
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

Adds `read_log_file() to read previous log object into list of tibbles object that you can more easily subset/filter. Addresses the first part of #162. Fixes #162.

### CODE REVIEW

**Comment by parmsam-pfizer** [comment]
Looks like the Ubuntu 3.6 image is also failing for this one now.

File: R/read_log_file.R
Comment (bms63) [inline R/read_log_file.R:1]: Follow Nic's PR on making it optional https://github.com/pharmaverse/logrx/pull/197

**Comment by bms63** [comment]
@parmsam-pfizer add yourself to the Description File!!

**Review by bms63 — COMMENTED** [review]
I was thinking more on the purpose of this update.  We are trying to extract and parse the contents of the to do some sort of analysis.

`read_log_file` might be a bit misleading...what about `extract_log_contents` , `parse_log_contents` or something else??

**Comment by parmsam-pfizer** [comment]
> 

The function is based on the function name scheme used in [readr](https://readr.tidyverse.org/reference/index.html). Maybe we can discuss alternate names at our next meeting. I'll work on the other edits in the meantime.

File: R/read_log_file.R
Comment (parmsam-pfizer) [reply R/read_log_file.R:1]: Thanks for the suggestion. 😄 Added condition based on @nicholas-masel code. Please let me know if I missed something with that optional change.

**Comment by parmsam-pfizer** [comment]
@nicholas-masel @bms63 
Made changes based on our meeting earlier today. Please feel free to merge PR if it looks good. 🚀

**Review by bms63 — APPROVED** [review]
LGTM! @nicholas-masel or @kodesiba any last words?!?

**Pull request merged**